NocillaTests/Stubs/LSStubRequestSpec.m

```diff
@@ -235,4 +236,77 @@
+        LSRegexMatcher *urlB = [[LSRegexMatcher alloc] initWithRegex:@"https://(.+)\\.example\\.com/(.+)/data\\.json".regex];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"GET" urlMatcher:urlB];
+
+        stubA.body = [[LSRegexMatcher alloc] initWithRegex:@"checkout my sexy body ([a-z]+?)".regex];
+        stubB.body = [[LSRegexMatcher alloc] initWithRegex:@"checkout my sexy body ([a-z]+?)".regex];
+
+        [stubA setHeader:@"Content-Type" value:@"application/json"];
+        [stubB setHeader:@"Content-Type" value:@"application/json"];
+
+        [[stubA should] equal:stubB];
+    });
+
+    it(@"is not equal with a different method", ^{
+        LSStubRequest *stubA = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"POST" url:@"google.com"];
+
+        [[stubA shouldNot] equal:stubB];
+    });
+
+    it(@"is not equal with a different url matcher", ^{
+        LSRegexMatcher *urlA = [[LSRegexMatcher alloc] initWithRegex:@"https://(.+)\\.example\\.com/(.+)/data\\.json".regex];
+        LSStubRequest *stubA = [[LSStubRequest alloc] initWithMethod:@"GET" urlMatcher:urlA];
+        LSRegexMatcher *urlB = [[LSRegexMatcher alloc] initWithRegex:@"https://github\\.com/(.+)/(.+)/".regex];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"GET" urlMatcher:urlB];
+
+        [[stubA shouldNot] equal:stubB];
+    });
+
+    it(@"is not equal with different headers", ^{
+        LSStubRequest *stubA = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+
+        [stubA setHeader:@"Content-Type" value:@"application/json"];
+        [stubB setHeader:@"Content-Type" value:@"application/json"];
+
+        [stubA setHeader:@"X-OMG" value:@"nothings compares"];
+        [stubB setHeader:@"X-DIFFERENT" value:@"to you"];
+
+        [[stubA shouldNot] equal:stubB];
+    });
+
+    it(@"is not equal with a different body", ^{
+        LSStubRequest *stubA = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+
+        stubA.body = [[LSStringMatcher alloc] initWithString:@"omg"];
+        stubB.body = [[LSStringMatcher alloc] initWithString:@"different"];
+
+        [[stubA shouldNot] equal:stubB];
+    });
+
+    it(@"is equal with a nil bodies", ^{
+        // Body is special in that it's the only propery that may be nil,
+        // and [nil isEqual:nil] returns NO.
+
+        LSStubRequest *stubA = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+        LSStubRequest *stubB = [[LSStubRequest alloc] initWithMethod:@"GET" url:@"google.com"];
+
+        stubA.body = nil;
+        stubB.body = nil;
+
+        [[stubA should] equal:stubB];
+    });
+});
```

Comment:
Oh man, thanks for the tests for `-isEqual:`. As I was reading the implementation, I found myself worried about how complex they were! Very reassuring to have this here.